### DISCUSSION TITLE
How to set focus on textbox when a view is shown

### DISCUSSION THREAD

**Discussion by simon52656** [AvaloniaUI/Avalonia#14655] [Q&A]
In my main user control I have several popup views/user controls for inputting info into my tool, these views are much smaller than the main view and the border greys out the background so you can still see what's going on but cant interact with it, 

![image](https://github.com/AvaloniaUI/Avalonia/assets/160526169/b89299e4-11a9-4a3a-aeed-f21e5f8a481b)

Every time the NewView is shown/hidden the NewViewModel is recreated so and this reset's the content,

In NewView.axaml.cs constructor I bind to the AttachedToVisualTree event to then focus the first textbox when the view is displayed. This works for the first time the view is displayed seen as the visual tree is only being created once, and the reused every other time the view is displayed.

![image](https://github.com/AvaloniaUI/Avalonia/assets/160526169/b568bb68-8aac-4275-b149-4d6db1070dad)

How can I go about making the textbox focused every time I open the view? 
Is there a way to remove the "NewView" from the visual tree when it's closed/hidden so it gets re-added every time?

I've already spent too long on this and hoping someone else knows something I don't :-) 

Edit: 
Replaced code snippets with screenshots as layout was unreadable

**Comment by workgroupengineering** [comment]
See https://github.com/AvaloniaUI/Avalonia.Xaml.Behaviors/blob/0c2ca085de718162146e5e68bb364f381e479532/src/Avalonia.Xaml.Interactions.Custom/FocusOnAttachedToVisualTreeBehavior.cs
